servicebroker/lib/routes/binding.js

```diff
@@ -70,53 +71,51 @@ module.exports = function(app, settings, catalog) {
                 var statusCode = response.statusCode;
                 logger.info("Api Server response", { status_code: statusCode, response: response.body });
 
-                switch(statusCode){
+                switch (statusCode) {
                   case 200:
                   case 201:
                     commitTransaction(t, res, statusCode, {credentials: {}});
                     return;
                   case 400:
-                    rollbackTransaction(t, res, statusCode, {error: response.body.error});
+                    rollbackTransaction(t, res, statusCode, { error: response.body.error });
                     return;
                   default:
                     rollbackTransaction(t, res, 500, {});
                     return;
                 }
 
               });
-            }).catch(function(error1) { //catch findorcreate
-              logger.error("Bind failed: create error", { error: error1 });
-              if (error1 instanceof models.sequelize.UniqueConstraintError) {
-                rollbackTransaction(t, res, 409, {});
-                return;
-              } else if (error1 instanceof models.sequelize.ForeignKeyConstraintError) {
-                rollbackTransaction(t, res, 404, { "description": messageUtil.getMessage("SERVICEINSTANCE_NOT_EXIST", { "serviceInstanceId": serviceInstanceId }) });
-                return;
-              }
-
-              rollbackTransaction(t, res, 500, {});
-            });
-          } else if (length > 1) { // an app has been bound to more than one service instance, this error should not exist
-            logger.error("Bind failed: duplicate bind", { app_guid: appId });
-            res.status(409).json({ "description": messageUtil.getMessage("DUPLICATE_BIND", { "applicationId": appId }) });
-            return;
-          } else if (length == 1) { // an app has been bound to a service instance
-            var bindingRecord = result[0];
-            if (bindingRecord.serviceInstanceId === serviceInstanceId) {
-              logger.error("Bind failed: app already bound", { app_guid: appId, serviceInstanceId: serviceInstanceId });
-              res.status(409).json({});
+            }
+
+          }).catch(function(error1) { //catch findorcreate
+            logger.error("Bind failed: create error", { error: error1 });
+            if (error1 instanceof models.sequelize.UniqueConstraintError) {
+              rollbackTransaction(t, res, 409, {});
```

Comment:
@qibobo  IMHO there should be some parity, we should not send for some 409 code and skip few.